source/rules/sync-compatibility.txt

```diff
@@ -0,0 +1,168 @@
+.. _device-sync-compatibility:
+
+==================================
+Device Sync-Compatible Permissions
+==================================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+When using :ref:`Device Sync (Flexible Mode) <sync>`, there are special
+considerations when using the permissions system.
+
+The older Partition-Based Sync uses a completely different permissions system.
+See :ref:`partition-based-rules-and-permissions` if you are using
+Partition-Based Sync.
+
+For a guide to setting up Flexible Sync with common permissions models, see the
+:ref:`flexible-sync-permissions-guide`.
+
+.. _sync-compatible-roles:
+
+Sync-Compatible Roles
+~~~~~~~~~~~~~~~~~~~~~
+
+If Device Sync (Flexible Mode) is enabled, an assigned :ref:`role <roles>` must
+be **sync compatible**. If the role is not sync compatible, but its "apply when"
+evaluated to true, other roles are not considered; access is denied.
+
+A role is not sync compatible if any of the following conditions are true:
+
+- ``document_filters.read`` or ``document_filters.write`` are undefined.
+- A document filter, insert, or delete expression references: a field that is
+  not a :ref:`queryable field <queryable-fields>`; an expansion other than
+  ``%%true``, ``%%false``, ``%%values``, ``%%environment``, or ``%%user``; or
+  the ``%function`` operator.
+- Field-level permissions are not boolean literals (``true`` or ``false``).
+- Field-level permissions are specified for the top-level ``_id`` field.
+
+
+.. _sync-compatible-expressions:
+
+Sync-Compatible Expressions
+---------------------------
+
+When Device Sync is enabled, expressions may only refer to the :ref:`queryable
+fields <queryable-fields>` of your data model. If a role refers to any other
+fields, it becomes Sync incompatible and cannot be used with Device Sync.
+
+Because a Sync-enabled App assigns roles at the start of a Sync session before
+any specific documents are queried, you can't refer to a document or its field
+values in an "apply when" expression.
+
+
+.. _sync-compatible-expansions:
+
+Sync-Compatible Expansions
+--------------------------
+
+When using Device Sync, some :ref:`expansions <expansions>` are not supported.
+The following table specifies which expansions are Sync compatible in either the
+"apply when" or rule expressions:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Expansion
+     - Can Use in "Apply When"?
+     - Can Use in Rule Expressions?
+   
+   * - :json-expansion:`%%true`, :json-expansion:`%%false`
+     - Yes
+     - Yes
+   
+   * - :json-expansion:`%%values`, :json-expansion:`%%environment`
+     - Yes
+     - Yes with an :ref:`important caveat <flex-sync-expansions-caveat>`
```

Comment:
Not all readers will understand the term "caveat". How do you feel about these alternatives?
- exception
- condition
- consideration

Reply:
Same applies to the other mention of caveat.